build/_build/DotNetSettingsExtensions.cs

```diff
@@ -88,4 +97,16 @@ public static DotNetTestSettings EnableTrxLogOutput(this DotNetTestSettings sett
             .SetLogger("trx")
             .SetResultsDirectory(resultsDirectory);
     }
+
+    /// <summary>
+    /// GitLab installs MSBuild in a non-standard place that causes issues for Nuke trying to resolve it
+    /// </summary>
+    public static MSBuildSettings SetMSBuildPath(this MSBuildSettings settings)
+    {
+        var vsRoot = Environment.GetEnvironmentVariable("VSTUDIO_ROOT");
+
+        return settings
+            .When(!string.IsNullOrEmpty(vsRoot),
+                c => c.SetProcessToolPath(Path.Combine(vsRoot, "MSBuild", "Current", "Bin", "MSBuild.exe")));
```

Comment:
The Windows Build image installs MSBuild in a weird location, so Nuke can't find it. Have to point it to the right place based on the presence of an environment variable. Not the nicest, but it works